### PR TITLE
Add votable coosys parsers

### PR DESCRIPTION
**Pull request by eteq**
This PR is the result of a collaborative effort at the IVOA <> Astropy sprint involving myself, @pllim, @astrofrog, @tomdonaldson, @tboch, and @bmatthieu3. This is the first in what will presumably be a few PRs to get "native" astropy objects out from VOTable's with the appropriate metadata.  This PR specifically gets `SkyCoords` from VOTables.

There are still three things to do before this is ready for review:

- [ ] Include proper motions when they are present in the table.  This is probably just looking for columns with the right set of UCDs and refs, but it still needs to be done.
- [ ] Make the two tables that are currently failing in the tests work.  For both of these there is a single set of ra/dec columns that are right for the  lone COOSYS, but there is no `meta.main` in the UCD.
- [ ] Decide which, if any, of these functions should be in the public API (as opposed to right now being hidden in `connect`)

For the second one, I'd appreciate thoughts on what the right thing is to do.  If there is no `meta.main`, do we just say it's not supported, or do we try to guess based on the presence of columns with the right UCD for the first COOSYS? (cc @astrofrog @tboch @tomdonaldson)

I'll make separate issues for the two follow-on tasks (adding a new votable reader that actually adds the `SkyCoord`'s to a table, and supporting timesys).

### CODE REVIEW

**Comment by eteq** [comment]
Oops... I should have done this as a draft since that's now a thing, but I forgot, so I just labeled it as a WIP.

**Comment by tboch** [comment]
>     * [ ]  Include proper motions when they are present in the table.  This is probably just looking for columns with the right set of UCDs and refs, but it still needs to be done.

I think that's an important addition to make, as it would be one of the first things I would check if I were to check correct behaviour of extract_skycoord_from_table
@eteq: what's the generic way to add proper motion components when creating SkyCoord?


>     * [ ]  Make the two tables that are currently failing in the tests work.  For both of these there is a single set of ra/dec columns that are right for the  lone COOSYS, but there is no `meta.main` in the UCD.

I have mixed feelings about what we should do.
Aladin Desktop uses a complex heuristic (based on UCDs, units and column names) to determine what are the fields making up the main coordinates in a table. We could go that way as well but I would not want this to delay a first implementation of this functionality.
Perhaps it's fine in a first time to have the tests to fail for these tables?

**Comment by eteq** [comment]
> what's the generic way to add proper motion components when creating SkyCoord?

There's two approaches that might work:

1. Use the names and assume that they match in the logical way: e.g. if the component is a declination, look for a `pm_dec` - this is basically just extending what's already there.  I think as it stands now that will work for all of the relevant definitions, although caveat of I'm not 100% sure whether it all works with the cos(dec) factor
2. Create representation objects.  That is, the `UnitSphericalRepresentation` will happily take a lat/lon and if you give it a `UnitSphericalDifferential` or `UnitSphericalCosLatDifferential`, we just need to map the UCDs to whether a column is a lat, lon, pm_lat, pm_lon, or pm_lon_coslat.  Then you can give that along with the `frame` to the `SkyCoord` initializer

Basically 1 is a minor perturbation on what's already in, but may not work, whereas 2 is more work but might be necessary depending on how the VO cos(dec) rules map onto the defaults for SkyCoord.

**Comment by eteq** [comment]
> Perhaps it's fine in a first time to have the tests to fail for these tables?

Good point. Presumably getting them to work is mainly a task of implementing the heuristic you're describing, so that's a fairly separable task than just getting this all working in the first place.

**Comment by pllim** [comment]
Haven't seen any progress since May, so I am moving the milestone, but keeping it open because it is something we need to revisit.

**Comment by hamogu** [comment]
Maybe this (and other related issues linking this PR) would make a good GSoC project if we can find mentors?

**Comment by bsipocz** [comment]
@hamogu - I've pinged you on slack

**Comment by pllim** [comment]
Moving milestone, as stated by @tomdonaldson during dev telecon.

**Comment by pllim** [comment]
Hello. I converted your PR to a draft PR due to #10706 .

**Comment by astrofrog** [comment]
@tomdonaldson @eteq - are you still interested in pushing ahead with this?  The 4.2 feature freeze is very soon (Oct 23rd) and it would be good to wrap this up.

**Comment by tomdonaldson** [comment]
> @tomdonaldson @eteq - are you still interested in pushing ahead with this? The 4.2 feature freeze is very soon (Oct 23rd) and it would be good to wrap this up.

Still interested, but unfortunately not for 4.2.

**Comment by eteq** [comment]
I agree with @hamogu that this is one of those "little" things that really do help! 

I'm inclined, though, to say we should leave it as at least "what we've got so far" until there is a rework done, @tomdonaldson. We should also think about whether we should avoid letting the perfect be the enemy of the good and shoot for something that's not perfect but is "at least better than what @hamogu has to do right now" as a target for 5.1. Might be a good thing to bring to the Python NAVO working group since that might offer a bit more enthusiasm to get it high enough in the priority?
